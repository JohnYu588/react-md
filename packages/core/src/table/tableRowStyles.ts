import { cnb } from "cnbuilder";
import { bem } from "../utils/bem.js";

const styles = bem("rmd-tr");

/** @since 6.0.0 */
export interface TableRowClassNameOptions {
  className?: string;
  disableHover?: boolean;
  disableBorders?: boolean;
  selected?: boolean;
  clickable?: boolean;
}

/** @since 6.0.0 */
export function tableRow(options: TableRowClassNameOptions = {}): string {
  const { disableHover, disableBorders, selected, clickable, className } =
    options;
  return cnb(
    styles({
      bordered: !disableBorders,
      hoverable: !disableHover,
      clickable,
      selected,
    }),
    className
  );
}
