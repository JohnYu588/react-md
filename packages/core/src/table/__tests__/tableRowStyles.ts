import { describe, expect, it } from "@jest/globals";
import { tableRow } from "../tableRowStyles.js";

describe("tableRow", () => {
  it("should be callable without any arguments", () => {
    expect(tableRow()).toMatchSnapshot();
  });
});
