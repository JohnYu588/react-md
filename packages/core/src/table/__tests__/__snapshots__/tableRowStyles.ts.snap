// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tableRow should be callable without any arguments 1`] = `"rmd-tr rmd-tr--bordered rmd-tr--hoverable"`;
